pkg/customresourcestate/config.go

```diff
@@ -158,21 +166,46 @@ type ConfigDecoder interface {
 	Decode(v interface{}) (err error)
 }
 
-// FromConfig decodes a configuration source into a slice of customresource.RegistryFactory that are ready to use.
-func FromConfig(decoder ConfigDecoder) ([]customresource.RegistryFactory, error) {
-	var crconfig Metrics
+// FromConfig decodes a configuration source into a slice of `customresource.RegistryFactory` that are ready to use.
+func FromConfig(decoder ConfigDecoder, r *discovery.GVKMaps) ([]customresource.RegistryFactory, error) {
```

Comment:
IMHO the factories generated by the discovery mechanism should be added elsewhere, this is solely meant for resources that can be deduced from the config.

Reply:
I added these here to ensure that the "resolved" `gvk`s were treated same as the other ones, and went through the same checks. I'm not sure what would be a better place to inject them from, do you have something in mind?

Reply:
https://github.com/kubernetes/kube-state-metrics/pull/1851/commits/8320aeddb5ae3d4eb16d9d7d98b3a060846de1a1#diff-bf91d16f16c99a97ad09eb6587f27d8fc075a68825d396e035c8100f864a7582R177